rules/travis.js

```diff
@@ -74,15 +75,23 @@ module.exports = ctx => {
 				});
 			}
 
-			const versions = Array.isArray(travis.node_js) ? travis.node_js : [travis.node_js];
+			let versions = Array.isArray(travis.node_js) ? travis.node_js : [travis.node_js];
+
+			versions = versions.filter(v => IGNORED_VERSIONS.indexOf(v) === -1);
 
 			for (const version of versions) {
 				if (DEPRECATED_VERSIONS.indexOf(version) !== -1) {
 					ctx.report({
 						message: `Version \`${version}\` is deprecated.`,
 						file
 					});
-				} else if (version !== 'node' && engine && !semver.satisfies(normalize(version), engine)) {
+				}
+			}
+
+			versions = versions.filter(v => DEPRECATED_VERSIONS.indexOf(v) === -1);
```

Comment:
Oh yes, you're right. Thanks for clarifying.

Although, maybe it might be a better idea to merge the two for..of loops and filter the `versions` array again after that for..of loop right before we enter the `if (engine)` block. I think it's just weird that we would do it in two loops while we can easily do it in one loop.